### PR TITLE
Adds option to only spawn if antag.

### PR DESCRIPTION
**Pull request by MrsTonedOne**
# The idea is to keep antag rollers from taking up job slots

They can late join in the remaining job slots if they want, or ghost, or whatever.

It will just send you to lobby other wise.

The fact of the matter is, this issue isn't gonna get solved administratively, and the only reason it is even an administrative issue is people stealing job slots. Antag Rollers are always going to exist, and with even the assistant slot being limited, we can't have them taking up job slots.

:cl:
add: You may now choose to only join at round start if you roll antag. See the game preferences screen for the option.
/:cl:

This PR brought to you by a smoking session between myself and @bobdobbington. The idea was partially his.

### CODE REVIEW

**Comment by LanCartwright** [comment]
you've gotta be kidding me.

**Comment by Firecage** [comment]
No.

**Comment by Iamgoofball** [comment]
:+1:

whats the problem with this PR

**Comment by kevinz000** [comment]
Isn't this an issue with player mindsets instead of code? All you're doing is making the problem worse as now everyone can just not play unless they are garunteed to be antag instead of having to roll.

**Comment by MrStonedOne** [comment]
and? they don't play anyways, they just do nothing, or the bare minimum to avoid issues. I would rather have 20 players who enjoy playing antag or not, then 60 players just rolling for antag.

**Comment by Firecage** [comment]
MSO? So what? If they round-start suicide or roundstart afk that frequently, then they can just be banned, jobbanned, or antag-banned.

If they want to play on the server, they should also play when they don't get antag. They shouldn't get a free pass when not getting it.

**Comment by Iamgoofball** [comment]
>then they can just be banned, jobbanned, or antag-banned.

>if people dont play when I want them to play they should be banned

selfish prick

**Comment by Firecage** [comment]
@Iamgoofball Please kindly fuck off with cherry picking everything. And No, if they don't get antag, they should still play. They shouldn't be able to cherry pick their rounds. They shouldn't only have to play when they get antag, and get a free pass when not getting antag.

**Comment by Iamgoofball** [comment]
>>then they can just be banned, jobbanned, or antag-banned.

>>if people dont play when I want them to play they should be banned

>selfish prick

**Comment by Archie700** [comment]
This is just "give antag rollers a free pass to observe if they don't get antag" the pr.

**Comment by Iamgoofball** [comment]
why do I have to play the game if I dont want to

**Comment by MrStonedOne** [comment]
Its better then them taking up slots.

**Comment by xxalpha** [comment]
I support this.

**Comment by Hatfish** [comment]
People will probably find a way to metagame this feature very quickly.

**Comment by Iamgoofball** [comment]
good thing we ban metagaming

**Comment by MrStonedOne** [comment]
Intentional.

If people announce that they enable this option, and somebody decides to meta that, as long as they are smart enough to not make it obvious that they meta'ed that to admins, i fail to see the issue.

I plan to implement a Don't ask; Don't tell policy on metaing be only antag. Admins will not actively attempt to seek out people doing that, but if its brought to their attention that somebody killed a tator and had no in game reason to think they were a tator, or the user outright admits to it, ban them for metagaming.

because honestly, if you don't want your antag status to get metagamed, play some fucking rounds.

**Comment by AlexCrimson** [comment]
Didnt really see the point at first, but when you put it like that, i suppose it makes sense. Well, as long as it reduces the number of suiciders, im fine with whatever.

**Comment by KorPhaeron** [comment]
This would literally kill the game

**Comment by KorPhaeron** [comment]
It's not selfish to expect people to take turns with different roles in an asymmetric PVP game

Nobody is forcing you to play if you don't want to. You just don't get to roll for traitor either.

**Comment by Incoming5643** [comment]
This pr is redundant, I already accidentally added this in #19381 

Just keep every job setting to none.

**Comment by GunHog** [comment]
It is rare for me to see good arguments on both sides (even if they are sometimes poorly written). On one end, there would be no need to put up with lost job slots, roundstart suiciders and the like. 

On the other, it could result in a drastic reduction of station population. Depending on how the code will work, could further reduce available game modes, as the antag-only players would probably ghost or avoid playing. On the long term, it could result in antag-only players filling up ghost roles more often, resulting in poor ghost role players. This could end up getting them removed.

I am worried that antag-only gamers could end up being entirely composed of the players whose non-antag playstyle is so poor that you cannot tell them apart form an antag.

I am seeing more bad than good coming from this one.

**Comment by KorPhaeron** [comment]
Unless you are killing yourself as a miner it is very unlikely you've "wasted a slot" anyway.

And most of our suicides are already greyshirts.

**Comment by Incoming5643** [comment]
Frankly we'd need a much larger playerbase to survive this sort of change.

Also lock it behind a server config, because downstreams ABSOLUTELY cannot survive this.

**Comment by KorPhaeron** [comment]
It's also kind of mind boggling that two admins, who spend a significant amount of time banning people for griefing, would enable and encourage players who literally only want to play when they have 0 rules and are allowed to grief.

**Comment by GunHog** [comment]
I am starting to see your point. Suiciders are the players that make a slight effort to be that obnoxious. Some would be perfectly okay playing their role even if they did not get antagonist. But, an _option_ that actively allowed guilt-free, completely safe antag-only play could be quite popular; Popular enough to leave a good chunk of the station barren, especially at lowpop.

I may be overestimating this, but this could result in a lot of empty stations.

**Comment by MrStonedOne** [comment]
Remove the meme pr tag @KorPhaeron

**Comment by yackemflam** [comment]
I, as a player who plays on this server, say that this is not a good idea.

I agree on Kor on this one.

**Comment by MrStonedOne** [comment]
greysuit slots are limited, and we see a lot of suicides there.

Clown, librarian, and mime slots are limited, and we see a non-trivial amount of suicides there.

There are random jobbers who suicide if not antag, so any slot, including security, engi, gen, rnd, etc can get used up.

and while i agree, most rounds, not every slot in every department is used up, there is a non-trivial amount of times somebody loses out on a job they would like to play, including greysuit, because of some antag roller.

***

Byond all of that, the people who don't suicide to avoid attracting attention to their antag hunting aren't good players, they just play enough to go thru the motions until they have an excuse to request the shuttle get called so they can get their re-roll.

I don't want those types of people playing, but we can't ban them, so this is the next best thing.

I'd rather have 20 players who enjoy playing the game when not antag than 60 players just going thru the motions barely playing at all.

**Comment by GunHog** [comment]
> I'd rather have 20 players who enjoy playing the game when not antag than 60 players just going thru the motions barely playing at all.

The problem I have with this, is that 60 players allow all the game modes. A 20 player (ready count) severely restricts the game modes open to us. I do not know about you, but I would like to play something other than Traitor(Chan) on a regular basis.

**Comment by MrStonedOne** [comment]
oh, no, gun_hog, that's the beauty of this pr, they are removed after that is checked!

**Comment by GunHog** [comment]
Well, I...umm...wow :/  That changes things a little...

**Comment by MrStonedOne** [comment]
think about it! I can't check that before, because in order to know if they rolled antag, game mode has to have already been selected.

**Comment by Incoming5643** [comment]
All that really means is that rounds can run with server populations that are too low to really support them. That's not great.

Also for the record at roundstart assistant slots AREN'T finite, the game needs a dump job to throw people who want to play but don't qualify for anything, and assistants are that job.

**Comment by ike709** [comment]
Try it for a few days. See if it totally murders station pop, or not.

**Comment by RemieRichards** [comment]
@Incoming5643 they are finite when they're set to finite in the config.

Ok so my opinion on this:
Uh... shit there's valid points on both sides.
If it can be proven, that the people who do this will ALWAYS suicide in their job slot, then yes, I could support this, but that's a hell of a thing to prove.

**Comment by calzilla1** [comment]
As a security player I would love the opertunity to metagame the hell out if this. Say if Smells was ONLY ANTAG and after a dozen rounds i find this out, then either i'd """random""" search him (her?) Every time I see it or put it on wanted every round for "suspicion"

**Comment by ike709** [comment]
Well you could also metagame someone who suicides every round except when they're antag. Hell, a single roundstart suicide from someone tells you that they either got a random job they don't like, or they just want antag.

**Comment by calzilla1** [comment]
This one's a bit more concrete.  When they chose antag, they won't show up on the manifest or any records, so i won't know who suicided but if they turn on ONLY ANTAG then every time they spawn they will be ANTAG

**Comment by KorPhaeron** [comment]
Also I feel like this is just going to attract a bunch of people from other servers to idle as ghosts while rolling for antag in the background of whatever else they're doing.

**Comment by Incoming5643** [comment]
@RemieRichards No, they **aren't**.

https://github.com/tgstation/tgstation/blob/master/code/controllers/subsystem/jobs.dm#L338

Any player that fails to qualify for any job they want who has their joblessrole set to BEASSISTANT will gain assistant at the end of job assignment. This happens regardless of how the configuration is set. You could have your configs set to disallow assistants entirely and you would still gain assistants by this mechanic. This has always been the case.

**Comment by RemieRichards** [comment]
@Incoming5643 really? I swear the assistant cap still prevents that.

**Comment by Incoming5643** [comment]
The population cap does but that's just a general thing, it's not tied to the job. It's an easy misconception though since by that point assistant is usually the only job left at that point.

**Comment by bobdobbington** [comment]
This PR is the best solution to a problem with very few good solutions. No amount of admin effort is going to solve the problem since admins are unable to differentiate "valid" roundstart suicides from antag rollers if the suicider isn't completely dumb and doesn't sabotage their own plausible deniability. As Kor himself has pointed out, people don't like being held hostage by their video games. Wasn't that why you quit mobas? What this PR does is enables a way to deal with this unavoidable issue without the people who actually do want to play the round suffering. This functionality needs to be at the very least tested unless someone's got a solution that doesn't force an impossible task on admins and acknowledges the reality of the situation.

**Comment by Incoming5643** [comment]
The best solution is to codify antag tokens and let people gain them by properly playing lots of rounds so they can get an assured antag once in a great while and stymie their horrible antag rolling lust the rest of the time.

I've been sitting on doing that for like, years.

**Comment by MrStonedOne** [comment]
That won't fix the issue @Incoming5643 because then you still have a bunch of players that only go thru the motions and aren't good players or crew members.

**Comment by Incoming5643** [comment]
I don't see how incentivizing people to not play is worse than incentivizing them to at least put in an effort for something they theoretically should want to do for free.

**Comment by AlexCrimson** [comment]
Antag Tokens are pretty controversial. If you start discussing them here, i doubt you will ever reach a conclusion.

Anyone unclear about how serious the suicider issue is, please go observe a few rounds from the beginning. You will be able to see just how many people kill themselves. Ive observed rounds that had 5 people suiciding within the first 5-10 minutes. Its hard to find an excuse for them when they run up to the HoP office to gib themselves then brag in deachat about how they started a mass suicide meme.

If it was limited to Assistants then i wouldnt care. However when the Virologist or a Scientist feels the needs to suicide in the central hallway, leaving their ID just laying around for anyone, AND depriving actual decent players of a job slot, i have no sympathy for them.

**Comment by bobdobbington** [comment]
>never accept the loss What happened to "you play in my universe, normie" coderbus is kill tbqh smh fam.

**Comment by Incoming5643** [comment]
>Antag Tokens are pretty controversial. If you start discussing them here, i doubt you will ever reach a conclusion.

Oh no he's seen through my gambit

**Comment by KorPhaeron** [comment]
@AlexCrimson 

Surely we should be banning those players if they've having a negative impact on the game instead of giving them what they want

@bobdobbington 

Yes, I quit mobas. Entirely. I didn't get a special deal where I only got to play when I was winning.

If people don't find the game fun, they don't have to play it. But they shouldn't be allowed to only play the most coveted role.

@MrStonedOne 

At least mediocre players provide fodder for the antags. 


Grief lottery is dumb, but the only way to fix that is to make station jobs more fun. However that's a fucking massive amount of work (it took me half a year to make lavaland, and that only keeps three civilians per round happy) so I doubt every job getting that treatment is gonna happen.

**Comment by KorPhaeron** [comment]
Or change the server rules of the server so the grief lottery isn't so extreme because right now we have "you smashed windows as non antag, enjoy your ban" as civillian vs "do literally whatever you want, silently kill everyone for no reason" as traitor.

Of course people are going to suicide when they get banned for breaking windows (yes this has happened repeatedly!) or even yelled at for talking to their friends too much (my universe, normie), leaving them locked into garbage """jobs""" while a handful of players each round just get to grief like crazy.

**Comment by AlexCrimson** [comment]
Hey, if you want to make a policy to stop the suiciders, then by all means do so mr headmin.

**Comment by GunHog** [comment]
Hey, if this IS passed, make sure the player is _ghosted_, not returned to lobby. Being able to roll for antag AND choose my job if I did not get antag is too enticing.

**Comment by KorPhaeron** [comment]
We do ban suiciders already @AlexCrimson 

Two people got perma antag banned recently for suiciding every round on both servers when not antag then immediately going ash walker, and one of them was permad after evading it

**Comment by PKPenguin321** [comment]
I would only use this like once or twice at the end of the day after playing a whole bunch of regular rounds. By using it like that it has minimal affect on the game and is a net positive. People that exclusively use this and never play a regular round will often not interact with the round at all and likely already don't (see: Join as assistant, immediately suicide). I doubt many people would turn this on and leave it on as it makes it so you essentially can't play the game most of the time. Those who would use this all the time likely already don't play the game very much.

Tl;dr: This will have minimal impact on the game. Most people won't use it exclusively, those that would already don't play because they just suicide every round.

This deserves to be extensively testmerged by the way

**Comment by AnturK** [comment]
Sending back to lobby seems kinda counterproductive here since usually that just means another roll.

I have absolutely no idea how will this affect playerbase, i'd suggest testmerge for a week then posting a statistic of people who enabled/not enabled this and in how many rounds.

**Comment by MrStonedOne** [comment]
This can't be test merged in it's current state, i'm gonna stash this commit and re-do it without the save file changes, add a config, and make it ghost them, (all extensively harder to do) and then we can test merge or just normal merge it (since it's a config)

I'll keep it attached to this pr to keep the comments, don't worry.

**Comment by AlexCrimson** [comment]
@KorPhaeron Its great that you ban them, but its still an unofficial policy and more of a rule 1 thing, right? Making it an official rule so that everyone is aware seems like it would be more effective than shadowbanning people.

**Comment by Incoming5643** [comment]
For the record it'd probably be easier to do with joblessrole and a check at the beginning of job assignment.

**Comment by MrStonedOne** [comment]
i tired that incoming, but you might get antag and still want to decide what happens when your job preference isn't available.

**Comment by Shapsy** [comment]
The only use this would have is as a honeypot to antagban the people who enable it, if you're not willing to play as the punching bag for other people, you don't get to play antag and ruin other people's rounds, plain and simple.

**Comment by drynwyn** [comment]
Some rounds allow latejoiners to be antagonists; independent of all other problems with this it gives people two antag rolls- one for whether or not they spawn in, and another for when they latejoin.

**Comment by Alek2ander** [comment]
A lot of stuff is tied to roundstart population, like nukeops points or blob stuff.

**Comment by Jalleo** [comment]
Uh guys you may all be forgetting the core issues forcing this.
One lavaland roundstart has to become a ready thing not observer.
Two everyone enjoys the game in their own way dont force the ones that feel like antag time to sit through a round of librarian or lawyer when they would rather go to sleep instead.
Three this is all caused because we knew we were eaeily losing players between rounds.
Four all solutions are not perfect we cant pick by semantics just what would be best for EACH individual player policy be damned.
Five if this gets abused for antag roles by specific players all the time we as coders can now document who they are and ask them to talk to our maintainers so we understand why they do it.
Six certain parts of this game are now so stale to old players they are repulsed by it and there is almost nothing we can do to bring the joy back to that without positive encouragement instead of indirect discouragement.
Seven this is a open source codebase on a awful game engine we cannot be wizards and make perfection.
Eight we cant even make mapping easy for our mappers to let them easily create new maps we have ways to reformat them and everything but we cant even develop a tool that makes mapping a one evening easy task.
That last point is kinda off topic but should be understandable for everyone who knows how strange it is to learn how to map in byond with SS13.
Either we go with this douchebag PR or we go with a polished turd PR your choice sheeple.

**Comment by Core0verload** [comment]
Where is MEME PR tag when we need it?

**Comment by Niknakflak** [comment]
It's all over your github account

**Comment by optimumtact** [comment]
With the design lead and headcoder against this and a train of thumbs down I suggest you stop making pr's while high.

**Comment by Iamgoofball** [comment]
REDDIT VOTES NOW MATTER GUYS

BREAK OUT THE ALTS

**Comment by xxalpha** [comment]
Citrusy tyranny. :tangerine:

**Comment by peoplearestrange** [comment]
How about we do the HG method. Why don't we FORCE TRAIL this for a week. Much like harmbots or spacelaw removal. Some will work and some won't, and we honestly won't know the full extent of the problems caused without trying it.

**Comment by AlexCrimson** [comment]
Separation of code and server. If Oranges doesnt want to fix the issue, just make a policy to stop it instead.

**Comment by PKPenguin321** [comment]
was this ever even testmerged? it deserved at least that

**Comment by KorPhaeron** [comment]
This version wasn't test mergeable anyway

**Comment by Dawgas** [comment]
I don't like how everyone's so fucking stupid that they don't see how this could be used to pick out everyone looking to roll for antag.

This is honestly just bait for bans - but then what kind of playerbase would be left?

The fact that nobody's agreeing with it just to be able to identify a group of people being blamed for the low quality of the game is just disgusting.

**Comment by Archie700** [comment]
After the first few days no one will fall for the bait.

**Comment by Dawgas** [comment]
do it like how VAC does it

ban people after like a month with some arbitrary metric so you don't tag everyone trying the new feature but get the shitlers, and then realize that "oh shit they weren't the problem oh well" or god knows what the actual use case would be.

**Comment by Davdichan** [comment]
Well if the policy discussions are anything to go by just play security and you can murder anyone who touches you no matter how you instigate it. Free valids
